back-end/www/models/model_operations/answer_operations.py

```diff
@@ -0,0 +1,390 @@
+
+    db.session.delete(answer)
+    db.session.commit()
+
+
+def exam_gold_standard(location_id, land_usage, expansion):
+    """
+    Check the quality of the answer in comparison with the gold standard.
+
+    Parameters
+    ----------
+    location_id : int
+        ID of the location.
+    land_usage : int
+        User's answer of judging if the land is a farm or has buildings.
+        (check the answer table in model.py for the meaning of the values)
+    expansion : int
+        User's answer of judging the construction is expanded.
+        (check the answer table in model.py for the meaning of the values)
+
+    Return
+    ------
+    int
+        Result of the checking.
+        0 means no gold standard exists.
+        1 means passing the gold standard test.
+        2 means failing the gold standard test.
+    """
+    # gold_standard_status 0 means that the answer is a gold standard
+    gold_answer = Answer.query.filter_by(gold_standard_status=0, location_id=location_id).first()
+
+    # If the gold answer doesn't exist
+    if gold_answer is None:
+        result = 0
+        return result
+
+    # If the gold answer exists, check the correctness
+    if gold_answer.land_usage == land_usage and gold_answer.expansion == expansion:
+        result = 1
+    else:
+        result = 2
+
+    return result
+
+
+def is_answer_reliable(location_id, land_usage, expansion):
+    """
+    Check if the answer to a location is reliable.
+
+    Before submitting the answer to the DB,
+    pass the target "good" answer candidate's information to check if another "good" answer candidate exists,
+    and have the same answer body.
+
+    The "good answer candidate" means the user passes the gold standard test when submitting it.
+
+    Parameters
+    ----------
+    location_id : int
+        ID of the location.
+    land_usage : int
+        User's answer of judging if the land is a farm or has buildings.
+        (check the answer table in model.py for the meaning of the values)
+    expansion : int
+        User's answer of judging the construction is expanded.
+        (check the answer table in model.py for the meaning of the values)
+
+    Return
+    ------
+    bool
+        Result of the checking.
+        True : Matches another good answer candiate.
+        False : No other good answer candidates exist or match.
+    """
+    # gold_standard_status 1 means that the answer passed the gold standard data quality test
+    good_answer_candidates = Answer.query.filter_by(gold_standard_status=1, location_id=location_id).all()
+
+    # If the gold candidate doesn't exist
+    if len(good_answer_candidates) == 0:
+        return False
+
+    for answer in good_answer_candidates:
+        if answer.gold_standard_status == 1 and answer.land_usage == land_usage and answer.expansion == expansion:
+            return True
+
+    return False
+
+
+def batch_process_answers(user_id, answers):
```

Comment:
I also need your help in explaining what this function is doing (step by step), let's chat this Wednesday during our regular meeting about this